CMakeLists.txt

```diff
@@ -52,6 +52,8 @@ option(IREE_ENABLE_RUNTIME_TRACING "Enables instrumented runtime tracing." OFF)
 option(IREE_ENABLE_COMPILER_TRACING "Enables instrumented compiler tracing." OFF)
 option(IREE_ENABLE_RENDERDOC_PROFILING "Enables profiling HAL devices with the RenderDoc tool." OFF)
 option(IREE_ENABLE_THREADING "Builds IREE in with thread library support." ON)
+set(IREE_RUNTIME_USE_FUTEX "AUTO" CACHE STRING
```

Comment:
this should be scoped only to the base/internal/ file instead of something global at this level, but I'm not sure I like the flag being exposed - if TSAN only works when futex is disabled then we should just condition IREE_PLATFORM_HAS_FUTEX on ` && !defined(IREE_SANITIZER_THREAD)`

KISS: configuration is complex enough and instructions not requiring multiple things to be set is better. It's also better not to rely on cmake configuration so that other build systems can also be used.

Reply:
I removed the CMake option.